pkg/bootstrap/bootstrap.go

```diff
@@ -0,0 +1,117 @@
+package bootstrap
+
+import (
+	"crypto/rand"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	types2 "github.com/obot-platform/obot/apiclient/types"
+	"github.com/obot-platform/obot/pkg/api"
+	"github.com/obot-platform/obot/pkg/api/authz"
+	"github.com/obot-platform/obot/pkg/gateway/client"
+	"github.com/obot-platform/obot/pkg/gateway/types"
+	"k8s.io/apiserver/pkg/authentication/authenticator"
+	"k8s.io/apiserver/pkg/authentication/user"
+)
+
+const bootstrapCookie = "obot-bootstrap"
+
+type Bootstrap struct {
+	token, serverURL string
+	gatewayClient    *client.Client
+}
+
+func New(serverURL string, c *client.Client) (*Bootstrap, error) {
+	token := os.Getenv("OBOT_BOOTSTRAP_TOKEN")
+
+	if token == "" {
+		bytes := make([]byte, 32)
+		_, err := rand.Read(bytes)
+		if err != nil {
+			return nil, fmt.Errorf("failed to generate random token: %w", err)
+		}
+
+		token = fmt.Sprintf("%x", bytes)
+
+		// We deliberately only print the token if it was not provided by the user.
+		fmt.Printf("Bootstrap token: %s\n", token)
```

Comment:
Do we print it so that user can grab it from the log? if so I would suggest the log to be more descriptive.

Reply:
Fixed